src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,115 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_INITIAL_CAPACITY = 16;
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int threshold;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        int hash = Math.abs(Objects.hashCode(key));
```

Comment:
Could be redundunt since I used hash in multiple places and if I will move it to separete method, I will calculate hash more than once invoking that method every time. If you suggesting just changing `Math.abs(Objects.hashCode(key))` to separete method it will be shorten only for few chars but I will create the whole method for that in few lines.

But done since I removed redundant hash in Node.